framework/TSA/AutoCorrelation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Auto Correlation Transformation for Characterization of Time-Series Data.
+"""
+import numpy as np
+
+from utils import InputData, InputTypes, xmlUtils
+from .TimeSeriesAnalyzer import TimeSeriesGenerator, TimeSeriesCharacterizer
+
+
+# utility methods
+class AutoCorrelation(TimeSeriesGenerator , TimeSeriesCharacterizer):
```

Comment:
This guy isn't a generator, right?

Reply:
Correct, The generate function just returns zero in this case. However if I remove the generate function and the TimeSeriesGenerator from AutoCorrelation it will crash. Any suggestions around this or should I leave it as is?

Reply:
I see that Time series Characterizer and Generator are no longer required so this issue will be fixed.

Reply:
I think Characterizer should stay; Generator can go though.

Reply:
@dylanjm and I were talking, I think we need a different tool than the SyntheticHistory generator to capture results for characterization instead of generation. I think he wants to pursue a PostProcessor for this.